relayer/chains/substrate/event_parser.go

```diff
@@ -312,22 +365,33 @@ func (con *connectionInfo) MarshalLogObject(enc zapcore.ObjectEncoder) error {
 func (con *connectionInfo) parseAttrs(log *zap.Logger, attributes interface{}) {
 	attrs := attributes.(ibcEventQueryItem)
 
+	var err error
+
+	var height exported.Height
+	if height, err = parseHeight(attrs["height"]); err != nil {
+		log.Error("error parsing connection height: ",
+			zap.Error(err),
+		)
+		return
+	}
+	con.Height = height.GetRevisionHeight()
+
 	con.ConnID = attrs["connection_id"].(string)
 	con.ClientID = attrs["client_id"].(string)
 	con.CounterpartyClientID = attrs["counterparty_client_id"].(string)
 	con.CounterpartyConnID = attrs["counterparty_connection_id"].(string)
 }
 
-// parses the attributes of ibc core client type
+// parses the height from an input interface
 func parseHeight(i interface{}) (clienttypes.Height, error) {
 	height := i.(ibcEventQueryItem)
 
-	revisionNumber, err := strconv.ParseUint(height["revision_number"].(string), 10, 64)
+	revisionNumber, err := cast.ToUint64E(height["revision_number"].(float64))
```

Comment:
Please see other comment regarding type assertion to `float64` and then casting to `uint64`

Reply:
Same as previous point